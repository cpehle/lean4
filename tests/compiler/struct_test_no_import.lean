-- Test that @[struct] attribute works without explicit import
-- This file intentionally does NOT import Lean.Compiler.StructAttr

@[struct]
structure Point where
  x : Float
  y : Float

@[struct]
structure Complex where
  real : Float
  imag : Float

@[struct, packed]
structure PackedData where
  a : UInt8
  b : UInt32
  c : UInt8

def testPoint : Point := { x := 1.0, y := 2.0 }
def testComplex : Complex := { real := 3.0, imag := 4.0 }
def testPacked : PackedData := { a := 1, b := 100, c := 2 }

def main : IO Unit := do
  let p := testPoint
  let c := testComplex
  let pd := testPacked
  IO.println s!"Point: ({p.x}, {p.y})"
  IO.println s!"Complex: {c.real} + {c.imag}i"
  IO.println s!"Packed: a={pd.a}, b={pd.b}, c={pd.c}"
