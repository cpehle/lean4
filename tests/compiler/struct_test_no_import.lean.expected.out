Point: (1.000000, 2.000000)
Complex: 3.000000 + 4.000000i
Packed: a=1, b=100, c=2
